File: backend/internal/dto/reconciliation.go
package dto

type ReconciliationStartRequest struct {
	AccountID     string  `json:"account_id" binding:"required"`
	ActualBalance float64 `json:"actual_balance" binding:"required"`
	Date          string  `json:"date" binding:"required"` // YYYY-MM-DD
}

type ReconciliationConfirmRequest struct {
	AccountID string `json:"account_id" binding:"required"`
	Date      string `json:"date" binding:"required"` // YYYY-MM-DD
}

type ReconciliationResponse struct {
	Difference            float64               `json:"difference"`
	FormattedDifference   string                `json:"formatted_difference"`
	UnmatchedTransactions []TransactionResponse `json:"unmatched_transactions"`
	Suggestions           string                `json:"suggestions"`
	Status                string                `json:"status"` // match, mismatch
}

File: frontend/src/services/reconciliation.ts
import api from '../utils/api';
import type { Transaction } from './transactions';

export interface ReconciliationStartRequest {
  account_id: string;
  actual_balance: number;
  date: string;
}

export interface ReconciliationConfirmRequest {
  account_id: string;
  date: string;
}

export interface ReconciliationResponse {
  difference: number;
  formatted_difference: string;
  unmatched_transactions: Transaction[];
  suggestions: string;
  status: 'match' | 'mismatch';
}

export const reconciliationService = {
  async startReconciliation(data: ReconciliationStartRequest): Promise<ReconciliationResponse> {
    const res = await api.post('/reconciliation/start', data);
    return res.data.data;
  },

  async confirmReconciliation(data: ReconciliationConfirmRequest): Promise<void> {
    await api.post('/reconciliation/confirm', data);
  },
};
export default reconciliationService;
